dune/grid/io/file/printgrid.hh

```diff
@@ -0,0 +1,191 @@
+// -*- tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+// vi: set et ts=4 sw=2 sts=2:
+
+#include <fstream>
+#include <string>
+
+#include <dune/grid/common/mcmgmapper.hh>
+
+namespace {
+
+  template<int dim>
+  struct ElementDataLayout
+  {
+    bool contains (Dune::GeometryType gt)
+    {
+      return gt.dim()==dim;
+    }
+  };
+
+  template<int dim>
+  struct NodeDataLayout
+  {
+    bool contains (Dune::GeometryType gt)
+    {
+      return gt.dim()==0;
+    }
+  };
+
+  // Move a point closer to basegeo's center by factor scale (used for drawing relative to the element)
+  template <typename B, typename C>
+  C centrify (B basegeo, C coords, double scale) {
```

Comment:
This should probably use `const B& basegeo` to avoid needlessly copying the geometry around